planet/subscription_request.py

```diff
@@ -127,6 +127,13 @@ def catalog_source(
         planet.exceptions.ClientError: If start_time or end_time are not valid
             datetimes
     '''
+    for i in range(len(item_types)):
```

Comment:
This code errors out when I use multiple item-types:
```
planet-client-python$> planet subscriptions request-catalog --item-types REScene,PSScene --asset-types basic_analytic_b2 --geometry geometry.geojson --start-time 2022-02-02
Traceback (most recent call last):
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/bin/planet", line 11, in <module>
    load_entry_point('planet', 'console_scripts', 'planet')()
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/lib/python3.7/site-packages/click/core.py", line 1134, in __call__
    return self.main(*args, **kwargs)
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/lib/python3.7/site-packages/click/core.py", line 1059, in main
    rv = self.invoke(ctx)
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/lib/python3.7/site-packages/click/core.py", line 1665, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/lib/python3.7/site-packages/click/core.py", line 1665, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/lib/python3.7/site-packages/click/core.py", line 1401, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/jennifer.kyle/.pyenv/versions/planet-client-python-3.7.9/lib/python3.7/site-packages/click/core.py", line 767, in invoke
    return __callback(*args, **kwargs)
  File "/Users/jennifer.kyle/pl/planet-client-python/planet/cli/cmds.py", line 57, in wrapper
    func(*args, **kwargs)
  File "/Users/jennifer.kyle/pl/planet-client-python/planet/cli/subscriptions.py", line 270, in request_catalog
    filter=filter)
  File "/Users/jennifer.kyle/pl/planet-client-python/planet/subscription_request.py", line 133, in catalog_source
    asset_types[i])
IndexError: list index out of range
```

it is fine if there is just one item-type
```
planet-client-python$> planet subscriptions request-catalog --item-types REScene --asset-types basic_analytic_b2 --geometry geometry.geojson --start-time 2022-02-02
{"type": "catalog", "parameters": {"item_types": ["REScene"], "asset_types": ["basic_analytic_b2"], "geometry": {"type": "Polygon", "coordinates": [[[-58.708965, -11.6562], [-58.66734, -11.6562], [-58.66734, -11.626594], [-58.708965, -11.626594], [-58.708965, -11.6562]]]}, "start_time": "2022-02-02T00:00:00Z"}}
```

Reply:
Ah, yes I wrote this so you'd have to explicitly write out the asset type you wanted for a given item type. Sure, your example would be fine, but what if you want `item_type_1`, `item_type_2`, `item_type_3`, where `item_type_1` and `item_type_2` are both compatible with `asset_type_a` and `item_type_3` is only compatible with `asset_type_b`. Would you want it to be able to parse something like

```console
❯ planet subscriptions request-catalog \
        --item-types item_type1,item_type_2,item_type_3 \    
        --asset-types asset_type_a, asset_type_b \
        --geometry ../jsons_for_tests/aoi.geojson \
        --start-time 2022-01-01
```

Reply:
Ah, I see your comment in Slack. As you pointed out, [the docs](https://developers.planet.com/docs/subscriptions/source/#validation) claim that a subscription will only be successfully made if one `item_type` is provided. So, as I mention and show in my [comment below](https://github.com/planetlabs/planet-client-python/pull/905#issuecomment-1499699397), I've raised a `ClientError` if multiple `item_types` are provided.